hardware/function_generators.py

```diff
@@ -38,44 +138,46 @@ class Agilent_33250A():
             the 10 MHz reference clock.
         duty_cycle (float): The duty cycle of the square wave form in percent.
     """
+
     def __init__(self, visa_search_term):
-        rm = visa.ResourceManager()
-        self.inst = rm.open_resource(visa_search_term)
+         super(Agilent_33250A, self).__init__(visa_search_term)
 
-    def identify(self):
-        """
-        Returns:
-            str: the response from the ``*IDN?`` GPIB query.
-        """
-        return self.inst.query('*IDN?')[:-1]
 
+    #must redefine properties in subclasses
     @property
     def frequency(self):
         f = self.inst.query('FREQ?')
-        return float(f)
+        return float(f) * u.hertz
+
+    @property
+    def volt(self):
+        return float(self.inst.query("VOLT?")) * u.volt
 
     @frequency.setter
     def frequency(self, val):
+
+        #max and min values taken from user manual
+        if val<1 * u.microhertz:
+            raise ValueError("Minimum frequency is 1µHz")
+        elif((waveform_name == "SIN" or waveform_name=="SQU") and

```

Comment:
Strings are fragile. Consider using [enums ](https://docs.python.org/3/library/enum.html)instead to represent `waveform_name`s.